src/StandardReactiveSocket.cpp

```diff
@@ -195,6 +211,23 @@ void StandardReactiveSocket::createResponder(
     ConnectionAutomaton& connection,
     StreamId streamId,
     std::unique_ptr<folly::IOBuf> serializedFrame) {
+  debugCheckCorrectExecutor();
+
+  if (streamId != 0) {
+    if (nextStreamId_ % 2 == streamId % 2) {
```

Comment:
this chunk of code will be moved to a separate class. I already started working on it.